### PR TITLE
Add vistrans code to eMPRess

### PR DESCRIPTION
**Pull request by justinjiang1212**
@ssantichaivekin Where should we put the vistrans code?

### CODE REVIEW

**Comment by ssantichaivekin** [comment]
I think putting it there is good :). On a side note, please use `snake_case` for python variable names, function names, and module names (folder names). `UpperCamelCase` are reserved for classes. (see [Google's python style guide](http://google.github.io/styleguide/pyguide.html#316-naming))

File: empress/vistrans/ReconRenderer.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
+import RenderClasses
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:1]: not a class, please use `snake_case` :)

File: empress/vistrans/ReconRenderer.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    logical_row_counter = 0
+    for leaf in host_tree.leaves:
+        leaf.logicalRow = logical_row_counter
+        logical_row_counter += 1
+    
+    #helper function to assign row values, postorder traversal
+    computetHostNodeLogicalPositions_helper(host_tree.rootNode)
+
+    
+def computetHostNodeLogicalPositions_helper(node):
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:37]: please use snake case for function names :)

File: empress/vistrans/RenderClasses.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+from enum import Enum
```
Comment (ssantichaivekin) [inline empress/vistrans/RenderClasses.py:1]: rename to use lower case I think.

File: empress/vistrans/ReconRenderer.py (hunk 2 above)
Comment (TrentonJW) [reply empress/vistrans/ReconRenderer.py:37]: Hey could you make me a collaborator to this repo?

Comment (ssantichaivekin) [reply empress/vistrans/ReconRenderer.py:37]: Yes.

**Comment by ssantichaivekin** [comment]
Instead of creating a pull request to `master` branch, please create a pull request to `topoSort` branch so that the changes from `topoSort` to master is not displayed twice.

File: empress/vistrans/ReconRenderer.py [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+    ### Here's what needs to happen here:
+    ### The logicalCol of each Node is just its order.
+    ### The logicalRows are computed as follows:
+    ###     1.  For leaves, we iterate through the list of leaves for the tree from left (top)
+    ###         to right (bottom) and make the logicalRow the indices 0 to numLeaves-1
+    ###     2.  Moving up the tree, the logical row of a node is the midpoint (a float) of 
+    ###         the logical rows of its two children.
+
+
+def computeParasiteNodeLogicalPositions(parasite_tree, host_tree, recon_map):
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:65]: Please use snake case function names.

File: empress/vistrans/ReconRenderer.py [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+        node.logicalRow = host_tree.nameToNode[event.hostName].logicalRow
+        
+    
+    ### Here's what needs to happen here:
+    ### The logicalCol of each Node is just its order.
+    ### The logicalRows are computed as follows:
+    ###     1.  For each node v of the parasite tree, its logicalRow is the same
+    ###         as the logicalRow of the host node to which it is mapped.  That's it!
+
+def computeHostNodeActualPositions(host_tree, x_min, x_max, y_min, y_max, x_offset=TREE_OFFSET[0], y_offset=TREE_OFFSET[1]):
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:92]: Please use snake case function names.

File: empress/vistrans/ReconRenderer.py [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+
+    #update actual positions
+    for node in host_tree.allNodes:
+        if node.root:
+            node.xcoord = x_min + x_offset
+            node.ycoord = node.logicalRow * y_unit + y_offset
+        else:
+            node.xcoord = node.logicalCol * x_unit + x_offset
+            node.ycoord = node.logicalRow * y_unit + y_offset
+
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:116]: Remove extra spaces.

File: empress/vistrans/ReconRenderer.py [hunk 7]
```diff
@@ -0,0 +1,418 @@
+import RenderClasses
+import tkinter as tk
+
+
+NODE_RADIUS = 5
+TREE_OFFSET = [20, 30]
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:6]: Could you explain in a comment what the two numbers mean? What is the tree offset, and why are there two values?

File: empress/vistrans/ReconRenderer.py [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+
+    n = 10
+    if up:
+        points = [x_pos, y_pos - n, x_pos - n, y_pos + n, x_pos + n, y_pos + n]
+    else:
+        points = [x_pos, y_pos + n, x_pos - n, y_pos - n, x_pos + n, y_pos - n]
+
+    canvas.create_polygon(points, fill=color, width=20)
+
+### Host Tree
```
Comment (ssantichaivekin) [inline empress/vistrans/ReconRenderer.py:281]: Please move examples and/or tests to a different file.

File: empress/vistrans/ReconRenderer.py (hunk 4 above)
Comment (justinjiang1212) [reply empress/vistrans/ReconRenderer.py:65]: this is our old code, we're just keeping it here for reference as we write new code that is compatible with the new data structures

Comment (ssantichaivekin) [reply empress/vistrans/ReconRenderer.py:65]: In that case, keep in in your computer but don't push it to the codebase.

**Comment by ssantichaivekin** [comment]
Pull request abandoned. (Prof Ran, Justin, Trenton will create a new one).

**Pull request closed without merging**